test/jax/test_math.py

```diff
@@ -36,7 +36,7 @@ def test_logdet():
     assert ld.dtype == jnp.complex64
     ldc = logdet_cmplx(A.astype(jnp.complex64))
     assert ldc.dtype == jnp.complex64
-    np.testing.assert_allclose(ld, ldc, rtol=1e-6)
+    np.testing.assert_allclose(ld, ldc, rtol=1.5e-6)
```

Comment:
Convenient adjustment! why?

Reply:
we change the rng implementation to the the partitionable one, 
and the tolerance in this test is so tight that it fails even if you just change the random seed:


```python
import netket as nk
import jax
import jax.numpy as jnp
import numpy as np    
from netket.jax import logsumexp_cplx, logdet_cmplx

#jax.config.update('jax_threefry_partitionable', True)
#k = jax.random.PRNGKey(1)

k = jax.random.PRNGKey(4)


A = jax.random.normal(k, (3, 4, 5, 5), dtype=jnp.float32)
ld = logdet_cmplx(A)
assert ld.shape == (3, 4)
assert ld.dtype == jnp.complex64
ldc = logdet_cmplx(A.astype(jnp.complex64))
assert ldc.dtype == jnp.complex64
np.testing.assert_allclose(ld, ldc, rtol=1e-6)
```
This runs in single precision and the absolute error when it fails is around 1e-5